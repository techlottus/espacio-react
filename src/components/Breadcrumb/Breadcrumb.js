import React, { useEffect } from "react";

export const Breadcrumb = React.memo(({ data, onItem, onBack }) => {
  const breadcrumbRef = React.createRef();

  useEffect(() => {
    breadcrumbRef.current.data = {
      textItems: data.textItems || [],
      textColor: data.textColor || "brand",
      // iconColor: data.iconColor || '',
      // itemFocusColor: data.itemFocusColor || '',
      icon: data.icon || "",
    };
  }, [data]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    breadcrumbRef.current.addEventListener("onItem", onItem);
    breadcrumbRef.current.addEventListener("onBack", onBack);
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-breadcrumb ref={breadcrumbRef}></lottus-breadcrumb>
    </>
  );
});
