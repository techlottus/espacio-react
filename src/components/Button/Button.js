

import React, { useEffect } from 'react'


export const Button = React.memo(({ data, onClick }) => {



  const buttonRef = React.createRef();
  
  useEffect(() => {
    buttonRef.current.data = {
      id: data.id || null,
      type: data.type  || 'primary',
      title: data.title || '',
      size: data.size || 'small',
      icon: data.icon || '',
      lyIcon: data.lyIcon,
      disabled: data.disabled,
      isExpand: data.isExpand,
      test: data.test || ''
    }
  }, [data]) // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    buttonRef.current.addEventListener('onClick', onClick);
  }, []) // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-button ref={buttonRef}></lottus-button>
    </>
  )
})


