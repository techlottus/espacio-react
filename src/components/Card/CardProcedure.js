import React, { useEffect } from "react";


import "./CardProcedure.scss";

export const CardProcedure = ({ data, text, price, onClick }) => {

  const cardRef = React.createRef();

  useEffect(() => {
    cardRef.current.data = {
      icon: data.icon || "",
      title: data.title || "",
      text: data.text || "",
      price: data.price || 0,
      color: data.color || "",
      disabled: data.disabled,
      ticket: data.ticket || "",
      date: data.date || "",
      isContent: data.isContent,
    };
  }, [data]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    cardRef.current.addEventListener('onClick', onClick);
  }, []) // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-card-icon style={{ width: "100%", height: "100%" }} ref={cardRef}>
        <div content="" className="container-content">
          <div className="text">
            <p>{text}</p>
          </div>
          <div className="price">
            <p>{price}</p>
          </div>
        </div>
      </lottus-card-icon>
    </>
  );
};
