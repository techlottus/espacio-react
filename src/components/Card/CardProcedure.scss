@import '../../App.scss';

.container-content {
  font-family: "Nunito", sans-serif;

  .text {
      padding-bottom: pixeltorem(31px);
    p {
      font-size: pixeltorem(14px);
      line-height: pixeltorem(19px);
    }
  }
  .price {
    p {
      font-weight: bold;
      font-size: pixeltorem(16px);
      line-height: pixeltorem(22px);
    }
  }
}
