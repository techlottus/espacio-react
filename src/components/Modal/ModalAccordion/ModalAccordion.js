import React, { useEffect, useRef, useState } from "react";
import "./ModalAccordion.scss";
import { Accordion } from "../../Accordion/Accordion";
import { useLocation } from "react-router-dom";
import { typeflowsProcedures } from "../../../constants/Procedures.constant";
import { sendInfoTM } from "../../../tagging/services/sendInfoTagManager";
import { closeIconRequirementsEquivalenceStudiesTag } from "../../../tagging/flows/procedures/equivalenceStudiesTag";
import { certificateStudyRequiTag } from "../../../tagging/flows/procedures/certificateStudyTag";
import { useSelector } from "react-redux";
import { accordionFormat } from "../../../helpers/accordionFormat";
export const ModalAccordion = ({ isShow, onClose, outside }) => {
  const location = useLocation();
  const { texts } = useSelector(state => state.texts);
  
  const modalRef = useRef();

  const [textsAccordion, setTextsAccordion] = useState({
    title: null,
    subtitle: null,
  });

  const [contentAccordion, setContentAccordion] = useState();
  const flow = location.pathname
    .replace("/procedures-flows/", "")
    .split("/")[0];
  useEffect(() => {
    switch (flow) {
      case typeflowsProcedures.equivalenceStudy:
        setTextsAccordion({
          title: texts?.procedures?.accordions?.equivalenceStudies,
          subtitle: "",
        });
        setContentAccordion(accordionFormat(texts.procedures.modalAccordion.equivalenceStudies));
        break;
      case typeflowsProcedures.certificateStudy:
        setTextsAccordion({
          title: texts?.procedures?.accordions?.studyRecord,
          subtitle: texts?.procedures?.accordions?.studyRecordSubtitle
        });
        setContentAccordion(accordionFormat(texts.procedures.modalAccordion.certificateStudy));
        break;
      default:
        break;
    }
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  const [isModal, setIsModal] = useState(false);

  useEffect(() => {
    setIsModal(isShow);
  }, [isShow]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    if (isShow) {
      document.body.style.overflowY = "hidden";
    } else {
      document.body.style.overflowY = "auto";
    }
  }, [isShow]);

  useEffect(() => {
    const handleClickOutside = (event) => {
      if (
        modalRef.current &&
        !modalRef.current.contains(event.target) && 
        window.innerWidth > 991
      ) {
        outside();
      }
    };
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  const handleModal = () => {
    return (
      <div className="modal-accordion">
        <div className="modalcontent" ref={modalRef}>
          <div className="modalheader">
            <div className="title">{textsAccordion.title}</div>
            <span
              className="material-icons icon"
              onClick={() => {
                onClose();
                switch (flow) {
                  case typeflowsProcedures.certificateStudy:
                    sendInfoTM(window, certificateStudyRequiTag, "close_x");
                    break;
                  case typeflowsProcedures.equivalenceStudy:
                    sendInfoTM(
                      window,
                      closeIconRequirementsEquivalenceStudiesTag
                    );
                    break;
                  default:
                    break;
                }
              }}
            >
              close
            </span>
          </div>
          <div className="modalbody">
            <div className="subtitle">{textsAccordion.subtitle}</div>
            <Accordion data={contentAccordion} />
          </div>
        </div>
      </div>
    );
  };

  return <>{isModal ? handleModal() : ""}</>;
};
