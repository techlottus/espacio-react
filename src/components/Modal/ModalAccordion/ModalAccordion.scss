@import '../../../App.scss';
@import "../Modal.scss";

.modal-accordion{
  @extend .modal-container;

  .subtitle{
    font-family: 'Nunito', sans-serif;
    padding-bottom: pixeltorem(15px);
  }
}
