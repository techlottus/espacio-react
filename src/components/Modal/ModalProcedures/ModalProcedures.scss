@import '../../../App.scss';
@import "../Modal.scss";

.modal-academic {
  @extend .modal-container;
  .modalbody {
    
    .academicprofile {
      margin: pixeltorem(16px) 0;
    }
    .infoacademic {
      font-family: "Nunito", sans-serif;
      font-size: pixeltorem(14px);
      line-height: pixeltorem(19px);
      padding-bottom: pixeltorem(16px);
      .infoacademictext {
        & .material-icons {
          padding-right: pixeltorem(11px);
        }
        &:last-child {
          padding-bottom: pixeltorem(16px);
        }
      }
    }
    .modalbtns {
      @media only screen and (min-width: map-get($grid-breakpoints, s)) {
        float: right;
      }
    }

    .contentfeedbackprocedures {
      font-family: "Nunito Bold", sans-serif;
      font-size: pixeltorem(18px);
      line-height: pixeltorem(24px);
    }
  }
}
