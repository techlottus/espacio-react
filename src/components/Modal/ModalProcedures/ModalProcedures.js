import React, { useEffect, useRef, useState } from "react";
import { Button } from "../../Button/Button";
import {
  btnModalAcademicContinue,
  btnModalAcademicExit,
} from "../../../constants/ProceduresAcademicHistory.constant";
import "./ModalProcedures.scss";
import { cardProfile } from "../../../constants/AccountStatus.constant";
import { Feedback } from "../../Feedback/Feedback";
import { CardProfile } from "../../Card/CardProfile";
import { feedbackProcedures } from "../../../constants/Procedures.constant";
import { useSelector } from "react-redux";
import { getImageOfAssets } from "../../../helpers/getImages";

export const ModalProcedures = ({
  isShow,
  onClose,
  onNext,
  infoModal,
  outside,
}) => {
  const [isModal, setIsModal] = useState(false);
  const { texts } = useSelector((state) => state.texts);

  const modalRef = useRef();

  const [profile, setProfile] = useState({
    ...cardProfile,
    image: getImageOfAssets(texts?.accountStatus?.images?.user),
  });

  const { profile: profileProcedure } = useSelector(
    (state) => state.procedures
  );

  useEffect(() => {
    if (infoModal.profile !== null) {
      const inforProfile = infoModal.profile;
      setProfile((state) => {
        return {
          ...state,
          name: inforProfile?.name,
          items: inforProfile?.items,
        };
      });
    }
  }, [infoModal.profile]); // eslint-disable-line react-hooks/exhaustive-deps

  const handleFeedback = () => {
    return (
      <div className="contentfeedbackprocedures">
        {infoModal.title} {texts?.procedures?.modalProcedures?.feedback}{" "}
        {infoModal.cost}
      </div>
    );
  };
  useEffect(() => {
    setIsModal(isShow);
  }, [isShow]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    if (isShow) {
      document.body.style.overflowY = "hidden";
    } else {
      document.body.style.overflowY = "auto";
    }
  }, [isShow]);

  useEffect(() => {
    const handleClickOutside = (event) => {
      if (
        modalRef.current &&
        !modalRef.current.contains(event.target) &&
        window.innerWidth > 991
      ) {
        outside();
      }
    };
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  const handleModal = () => {
    return (
      <div className="modal-academic">
        <div className="modalcontent" ref={modalRef}>
          <div className="modalheader">
            <div className="title">
              {texts?.procedures?.modalProcedures?.title}
            </div>
          </div>
          <div className="modalbody">
            {profileProcedure ? (
              <div className="academicprofile">
                <CardProfile data={profile} />
              </div>
            ) : (
              ""
            )}
            <div className="infoacademic">
              {infoModal.information.map((info, i) => {
                return (
                  <p key={i} className="infoacademictext">
                    {info}
                  </p>
                );
              })}
              {infoModal?.file && Array.isArray(infoModal.file)
                ? infoModal.file.map((doc, i) => {
                    return (
                      <p key={i} className="infoacademictext">
                        <span className="material-icons outlined">
                          description
                        </span>
                        {doc}
                      </p>
                    );
                  })
                : ""}
            </div>
            {infoModal.title ? (
              <div className="feedbackcontent">
                <Feedback data={feedbackProcedures} html={handleFeedback} />
              </div>
            ) : (
              <></>
            )}

            <div className="modalbtns">
              <div className="btnsecundary">
                <Button
                  data={{
                    ...btnModalAcademicExit,
                    isExpand: window.innerWidth < 991,
                  }}
                  onClick={() => {
                    onClose();
                  }}
                />
              </div>
              <div className="btnprimary">
                <Button
                  data={{
                    ...btnModalAcademicContinue,
                    isExpand: window.innerWidth < 991,
                  }}
                  onClick={() => {
                    onNext();
                  }}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  };

  return <>{isModal ? handleModal() : ""}</>;
};
