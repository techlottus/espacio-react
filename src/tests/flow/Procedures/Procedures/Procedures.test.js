import React from "react";
import { mount } from "enzyme";
import Procedures from "../../../../flow/Procedures/Procedures";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { act } from "react-dom/test-utils";
import { ProceduresContext } from "../../../../flow/Procedures/ProceduresContext";
import { BrowserRouter } from "react-router-dom";
import { ModalSolicitude } from "../../../../components/Modal/ModalSolicitude/ModalSolicitude";
import { CardProcedure } from "../../../../components/Card/CardProcedure";
import Sidebar from "../../../../components/Sidebar/Sidebar";
import { Button } from "../../../../components/Button/Button";
import { last } from "rxjs";
import { Breadcrumb } from "../../../../components/Breadcrumb/Breadcrumb";
import { typesRequestErrors } from "../../../../constants/error.constant";
import { ModalAccordion } from "../../../../components/Modal/ModalAccordion/ModalAccordion";
import { ModalProcedures } from "../../../../components/Modal/ModalProcedures/ModalProcedures";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

window.scrollTo = jest.fn();

let initState = {
  procedures: {
    requestInquiryAll: null,
    proceduresAll: null,
    profile: null,
    errors: {
      [typesRequestErrors.getInformationProcedures]: {
        isError: false,
        msg: null,
      },
    },
  },
  texts: {
    procedures: {
      phoneText: "666",
    },
  },
};

const historyMock = {
  push: jest.fn(),
  location: { state: { status: true } },
  listen: jest.fn(),
};

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <ProceduresContext.Provider>
      <Provider store={store}>
        <BrowserRouter>
          <Procedures history={historyMock} />
        </BrowserRouter>
      </Provider>
    </ProceduresContext.Provider>
  );
};

describe("Test <Procedures/>", () => {
  beforeEach(() => {
    store = mockStore(initState);
  });

  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });

  test("Should retrieve procedures information", () => {
    initState = {
        ...initState,
        procedures: {
          ...initState.procedures,
          proceduresAll: [
            {
              path: "/procedures-flows/study-record/information",
              icon: "description",
              type: "study-record",
              title: "Constancia de estudio",
              disabled: false,
              isContent: true,
              content: {
                text: "Documento con el que cada estudiante puede acreditar toda su trayectoria académica. En dicho documento se certifica oficialmente los estudios que una persona ha cursado con éxito.",
                price: "$480.00 - $590.00",
              },
            },
            {
              path: "/procedures-flows/certificate-study/information",
              icon: "star",
              type: "certificate-study",
              title: "Certificado de estudios",
              disabled: false,
              isContent: true,
              content: {
                text: "Documento donde se declara oficialmente que haz acreditado tu plan de estudios, este puede ser completo o parcial, incluye fechas, calificaciones, nombre de las unidades de aprendizaje, sellos y firmas de autoridades.",
                price: "$480.00 - $590.00",
              },
            },
          ],
        },
      };
    const wrapper = handleMount(initState);
    wrapper.update()
  });
  test("should close modal accordion", () => {
    const wrapper = handleMount(initState);

    act(() => {
      wrapper.find(ModalAccordion).props().onClose();
    });
    wrapper.update();
  });

  describe("Should click on Modal Procedures", () => {
    test("Should close modal procedures", () => {
      const wrapper = handleMount(initState);

      act(() => {
        wrapper.find(ModalProcedures).props().onClose();
      });

      wrapper.update();
    });

    test("Should handle request", () => {
      const wrapper = handleMount(initState);

      act(() => {
        wrapper.find(ModalProcedures).props().onNext();
      });
      wrapper.update();
    });
  });
});
