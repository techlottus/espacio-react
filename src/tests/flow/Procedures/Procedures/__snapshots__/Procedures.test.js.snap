// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Procedures/> should init component 1`] = `
<div
  className="procedures-container"
>
  <div
    className="navbar"
  >
    <lottus-navbar />
  </div>
  <div
    className="procedurelayout-container"
  >
    <div
      className="infocontent"
    >
      <div
        className="info-view"
      >
        <div
          className="infoview-container"
        >
          <div
            className="bread-info"
          >
            <lottus-breadcrumb />
          </div>
          <div
            className="info-title"
          >
            <h2 />
          </div>
          <div
            className="info-text"
          >
            <p />
          </div>
          <div
            className="info-price"
          >
            <p />
          </div>
        </div>
      </div>
    </div>
    <div
      className="formscontent"
    >
      <div
        className="forms-container"
      >
        <div
          className="forms-view"
        >
          <div
            className="forms-stepper"
          >
            <lottus-stepper />
          </div>
          <div
            className="forms-content"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
